pkg/v1/tkg/clusterclient/clusterclient.go

```diff
@@ -2309,28 +2309,8 @@ func (c *client) IsClusterRegisteredToTMC() (bool, error) {
 
 // VerifyCLIPluginCRD returns true if CRD exists else return false
 func (c *client) VerifyCLIPluginCRD() (bool, error) {
-	restconfigClient, err := c.GetRestConfigClient()
-	if err != nil {
-		return false, err
-	}
-	clusterQueryClient, err := capdiscovery.NewClusterQueryClientForConfig(restconfigClient)
-	if err != nil {
-		return false, err
-	}
+	return c.isCRDAvailableInAPIGroup("CLIPlugin", constants.TanzuCLIAPIGroupPath)
 
-	// Check if 'cliplugins' CRD is present or not
-	agent := &corev1.ObjectReference{
-		Kind:       "CustomResourceDefinition",
-		Name:       "cliplugins.cli.tanzu.vmware.com",
-		APIVersion: "apiextensions.k8s.io/v1",
-	}
-	var queryObject = capdiscovery.Object("CLIPluginCRDObject", agent)
```

Comment:
Ack, thanks, I originally went down that route but wasn't sure which one was preferred since there seemed to be a mix of approaches.

Not super familiar with the discovery pieces, but to confirm, would it look something like the following? (ie. should we include the resource name, not just the Group?)
```
func GroupResource(queryName, group string, resource string) *QueryGVR {
	return &QueryGVR{
		name:     queryName,
		group:    group,
		resource: nullString{IsSet: true, String: resource},
	}
}
```

Reply:
Ack, thanks for the pointer, will address this.

Reply:
Er, just to confirm, were you suggesting reworking the TKC discovery logic to use the same discovery package?

Reply:
> Er, just to confirm, were you suggesting also reworking the TKC discovery logic to use the same discovery package?

No, just to replace `Object` with `Group` query in this instance. I don't know what the impact of reworking TKC discovery logic tbh.

Reply:
Sounds good..probably a smaller blast radius, ha.

Reply:
Hmm. It's a bit of a pain to unit test because the existing code always loads the REST config (and I feel like writing tests for querying /api, etc. smells off). I think I might refactor this to be able to more easily inject a fake dynamic client so that I can use `discovery.NewClusterQueryClient` instead.

Reply:
Okay, I've tried to tweak it to use the Group query as you suggested.

One thing that bothered me was passing in a nil dynamic client... (if it's ever used by mistake, I think the unit test would catch it - it's not used in the discovery path as far as I can tell)

Reply:
Curious why you couldn't just keep this part?

```go
restconfigClient, err := c.GetRestConfigClient()
 if err != nil {
 	return false, err
 }
 clusterQueryClient, err := capdiscovery.NewClusterQueryClientForConfig(restconfigClient)
```

That would take care of creating the dynamic client. It's not used for this query, but probably just safer?

Reply:
Curious why you couldn't just keep this part?

```go
restconfigClient, err := c.GetRestConfigClient()
 if err != nil {
 	return false, err
 }
 clusterQueryClient, err := capdiscovery.NewClusterQueryClientForConfig(restconfigClient)
```

That would take care of creating the dynamic client. It's not used for this query, but probably just safer?

Reply:
Uh, mainly for ease of unit testing/dependency injection - it seemed a bit weird to have a client.discoveryClient but reinitialize one during the VerifyCLIPlugin() call. Thoughts?

Reply:
Perhaps a cleaner thing to do might be to allow injecting the dynamic client via a factory as well? Same way as the discovery client?

Reply:
Would this work for you to inject your own dynamic client? https://github.com/vmware-tanzu/tanzu-framework/blob/a1bd9830338d57a8c3f52f152e9994b5980f4e8b/pkg/v1/sdk/capabilities/discovery/discovery.go#L28

Reply:
latest diff has the patch I was thinking of to inject a dynamic client into the cluster client.